src/UI/Component/MainControls/Factory.php

```diff
@@ -0,0 +1,271 @@
+	 *     closed, i.e. removed from the Mainbar, via a Close Button. When the last
+	 *     Tool is closed, the tools-section is removed as well.
+	 *
+	 *   rivals:
+	 *     Tab Bar: >
+	 *       The Mainbar (and its components) shall not be used to substitute
+	 *       functionality available at objects, such as settings, members or
+	 *       learning progress. Those remain in the Tab Bar.
+	 *
+	 *     Content Actions: >
+	 *       "New item"-actions, the actions-menu (with comments, notes and tags),
+	 *       moving, linking or deleting objects and the like are NOT part of
+	 *       the Mainbar.
+	 *
+	 *     Personal Desktop: >
+	 *       The Personal Desktop provides access to services and tools and
+	 *       displays further information at first glance (e.g. the calendar).
+	 *       The Mainbar may reference those tools as well, but rather in form
+	 *       of a link than a widget.
+	 *
+	 *     Metabar: >
+	 *       Notifications from the system to the user, e.g. new Mail, are placed
+	 *       in Elements of the Metabar. The general direction of communication for
+	 *       the Mainbar is "user to system", while the direction is "system to user"
+	 *       with elements of the Metabar. However, navigation from both components
+	 *       can lead to the same page.
+	 *
+	 * context:
+	 *    - The Mainbar is used in the Standard Page.
+	 *
+	 * rules:
+	 *   usage:
+	 *     1: There SHOULD be a Mainbar on the page.
+	 *     2: If there is a Mainbar, it MUST be unique for the page.
+	 *
+	 *   composition:
+	 *     1: The bar MUST NOT contain items other than Bulky Buttons or Slates.
+	 *     2: The bar MUST contain at least one Entry.
+	 *     3: The bar SHOULD NOT contain more than five Entries.
+	 *     4: >
+	 *       Entries in the Mainbar MUST NOT be enhanced with counters or
+	 *       other notifications drawing the user's attention.
+	 *
+	 *   style:
+	 *     1: The bar MUST have a fixed width (desktop).
+	 *
+	 *   interaction:
+	 *     1: >
+	 *        Operating elements in the bar MUST either lead to further navigational
+	 *        options within the bar (open a slate) OR actually invoke navigation, i.e.
+	 *        change the location/content of the current page.
+	 *     2: Elements in the bar MUST NOT open a modal or window.
+	 *
+	 * ----
+	 *
+	 * @return  \ILIAS\UI\Component\MainControls\Mainbar
+	 */
+	public function mainbar(): Mainbar;
+
+
+	/**
+	 * ---
+	 * description:
+	 *   purpose: >
+	 *     A Slate is a collection of Components that serve a specific and singular
+	 *     purpose in their entirety. The purpose can be subsummed in one Icon/Glyph
+	 *     and one word, for Slates will act as elaboration on one specific concept
+	 *     in ILIAS.
+	 *
+	 *     Slates are not part of the content and will reside next to or over it. They
+	 *     will open and close without changing the current context.
+	 *     Accordingly, Slates depend on a component that toggles their visibility.
```

Comment:
I find it hard to understand if the triggerer and its properties is an external or internal thing to the slate. Who is taking care of icon/caption? Usually (like with buttons and popovers) the content container (popover) is separated from the triggerer. The current "Legacy Slate" example illustrates this unclear responsiblity imo. A triggerer (bulky button) has an icon/symbol and name/caption. But this is obtained from the slate object. What if they use different icons?

Reply:
We model it like this: The label and the icon(or glyph) belong to the slate, while the component that uses slates needs to define how these two are wrapped into a concrete triggerer (i.e. the button). We think, that the icon/glyph and label need to belong to the slate, because they are deeply connected to the content of the slate and thus can only be determined by the slates creator. The trigger on the other hand is a question of the context in which a slate is used and thus needs to be determined there. Does that help?

Reply:
Yes. I am not sure whether this could be reflected by a common interface accepted by the (bar) components, something that both, buttons and slates have to implement.